client/src/components/organisms/AdminRequestGroupList.tsx

```diff
@@ -0,0 +1,144 @@
+import { bindActionCreators, Dispatch } from "redux"
+import { gql, useQuery } from "@apollo/client";
+import { loadRequestGroups, setDisplayRequestGroups } from '../../data/actions/requestGroupsActions';
+import React, { FunctionComponent, useState } from "react";
+import { Button } from "../atoms/Button";
+import { connect } from "react-redux";
+import RequestGroup from '../../data/types/requestGroup';
+import RequestGroupTable from "../molecules/RequestGroupTable";
+import { RootState } from '../../data/reducers'
+import SearchBar from "../atoms/SearchBar";
+import SimplePageNavigation from "../atoms/SimplePageNavigation";
+
+interface StateProps {
+    requestGroups: Array<RequestGroup>,
+    displayRequestGroups: Array<RequestGroup>
+}
+
+interface DispatchProps {
+    loadRequestGroups: typeof loadRequestGroups,
+    setDisplayRequestGroups: typeof setDisplayRequestGroups
+}
+
+type Props = StateProps & DispatchProps;
+
+const AdminRequestGroupList: FunctionComponent<Props> = (props: React.PropsWithChildren<Props>) => {
+    // requestGroups are all the request groups that fit the search string
+    const [requestGroups, setRequestGroups] = useState<Array<RequestGroup>>(props.displayRequestGroups);
```

Comment:
Since we get `requestGroups` and `displayRequestGroups` in props, we probably don't need to duplicate the data in the components state.

My recommendation: 

1. Remove:
```
const [requestGroups, setRequestGroups] = useState<Array<RequestGroup>>(props.displayRequestGroups);
    // displayRequestGroups is a subset of requestGroups, and represents all the request groups on the current page
    const [displayRequestGroups, setDisplayRequestGroups] = useState<Array<RequestGroup>>(props.displayRequestGroups);
```

2. And change all references from `requestGroups` to `props.requestGroups` and `displayRequestGroups` to `props.displayRequestGroups`

3. Move the logic for getting a page of `props.displayRequestGroups` to props for `RequestGroupTable`
```
const handlePageChange = (newPage: number) => {
        setCurrentPage(newPage);
    };

    useQuery(query, {
        onCompleted: (data: { requestGroups: Array<RequestGroup> }) => {
            const displayRequestGroups = sortRequestGroupsAlphabetically(data.requestGroups.map(requestGroup => ({ ...requestGroup })));
            props.loadRequestGroups(data.requestGroups);
            props.setDisplayRequestGroups(displayRequestGroups);
        },
    });

    const onSearchStringChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setCurrentPage(1); // when search string changes, reset pagination
        let updatedRequestGroups = [];
        if (event.target.value.length > 0) {
            updatedRequestGroups = props.displayRequestGroups.filter(requestGroup => requestGroup.name == event.target.value);
        } else {
            // if no search string entered, return all results
            updatedRequestGroups = props.displayRequestGroups; 
        }
        props.setDisplayRequestGroups(updatedRequestGroups);
    }

    return (
        <span>
            // ...
            <RequestGroupTable requestGroups={props.displayRequestGroups.slice((currentPage - 1) * numGroupsPerPage,
                Math.min(
                    currentPage * numGroupsPerPage,
                    props.displayRequestGroups.length > 0 ? props.displayRequestGroups.length : Infinity
                ))} />
            // ...
        </span>
    );
``` 

Thoughts?

Reply:
@meganniu good call! made a few small changes to the suggested code, but this looks a lot cleaner :)
one thing I want to check:
- instead of `props.loadRequestGroups(data.requestGroups);`, I have `props.loadRequestGroups(displayRequestGroups);`, so basically I'm storing the sorted and filtered request groups into the redux store instead of all the groups returned by the server. is this going to be a problem? the reason I'm doing this is because whenever the search string changes, I reset `props.displayRequestGroups` to equal `props.requestGroups`, and if we store the raw list of request groups inside `props.requestGroups`, we'd have to do the sorting and filtering every time the search changed.

Reply:
Looks good, thanks for being so careful!!!